### PR TITLE
Detail product client

### PR DESCRIPTION
**Pull request by HoaiPEC**
## Related Tickets:
- [28522](https://edu-redmine.sun-asterisk.vn/issues/28522)

## Notes Impacted Areas:
- N/A.

## Attachments:
detail-product

### CODE REVIEW

**Comment by HoaiPEC** [comment]
ready

File: app/Http/Controllers/Clients/ProductController.php [hunk 1]
```diff
@@ -16,4 +16,16 @@ public function index()
...
         return view('client.products.index', compact(['categories', 'products']));
     }
+
+    public function detailProduct($id)
+    {
+        $product = Product::findOrFail($id);
+        $product_relates = Product::where('category_id', $product->category_id)->where('id', '<>', $id)
+            ->orderBy('updated_at', 'desc')->paginate(config('number-items.number_related'));
+        $rates = $product->rates();
+        $count_rate = $rates->count();
```
Comment (huyenttm-0901) [inline app/Http/Controllers/Clients/ProductController.php:26]:
```suggestion
        $countRate = $rates->count();
```

File: app/Http/Controllers/Clients/ProductController.php [hunk 2]
```diff
@@ -16,4 +16,16 @@ public function index()
 
         return view('client.products.index', compact(['categories', 'products']));
     }
+
+    public function detailProduct($id)
+    {
+        $product = Product::findOrFail($id);
+        $product_relates = Product::where('category_id', $product->category_id)->where('id', '<>', $id)
```
Comment (huyenttm-0901) [inline app/Http/Controllers/Clients/ProductController.php:23]:
```suggestion
        $productRelates = Product::where('category_id', $product->category_id)->where('id', '<>', $id)
```

File: resources/views/client/products/detail.blade.php [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                            <span>{{ '(' . $count_rate . ' ' . trans('clients.review') . ')' }}</span>
+                        </div>
+                        <div class="product__details__price">
+                            {{ ($product->price_discount) ? number_format($product->price_discount) : number_format($product->price) }}
+                            {{ config('number-items.unit') }}
+                        </div>
+                        <p>{{ $product->description }}</p>
+                        <div class="w-100">{{ $product->weight_available . ' ' . $product->unit . ' '. trans('clients.available') }} </div>
+                        <div class="col-md-12">
+                            <p style="color: #000;"></p>
```
Comment (huyenttm-0901) [inline resources/views/client/products/detail.blade.php:44]: css này

File: app/Http/Controllers/Clients/ProductController.php (hunk 1 above)
Comment (HoaiPEC) [reply app/Http/Controllers/Clients/ProductController.php:26]: e sửa rồi ạ

File: resources/views/client/products/detail.blade.php (hunk 3 above)
Comment (HoaiPEC) [reply resources/views/client/products/detail.blade.php:44]: e sửa rồi ạ

**Comment by HoaiPEC** [comment]
ready

File: app/Http/Controllers/Clients/ProductController.php [hunk 4]
```diff
@@ -16,4 +16,16 @@ public function index()
...
     }
+
+    public function detailProduct($id)
+    {
+        $product = Product::findOrFail($id);
+        $productRelates = Product::where('category_id', $product->category_id)->where('id', '<>', $id)
+            ->orderBy('updated_at', 'desc')->paginate(config('number-items.number_related'));
+        $rates = $product->rates();
+        $countRate = $rates->count();
+        $rating = ceil($rates->avg('rating'));
```
Comment (congnx-1020) [inline app/Http/Controllers/Clients/ProductController.php:27]: $productRelates = Product::where('category_id', $product->category_id)
     ->where('id', '<>', $id)
    ->orderBy('updated_at', 'desc')
    ->paginate(config('numer-items.number_related'));

File: app/Models/Product.php [hunk 5]
```diff
@@ -7,6 +7,7 @@
 class Product extends Model
 {
     protected $fillable = [
+        'id',
         'name',
```
Comment (congnx-1020) [inline app/Models/Product.php:11]: bo id ra ngoai

**Pull request merged**